### PR TITLE
Migrate query enhancement tests from functional repo to main

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description

P0 PR for migrating the two query enhancement tests with its utils from functional repo to main. Modify cypress test workflow to allow this.

Need to merge this functional repo PR with this: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1661

Built upon the comments in this PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8986
## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by abbyhu2000** [comment]
Test flow link: Running this PR workflow with functional repo PR:https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1661

All cypress tests pass: 
* ciGroup 1-9: cypress tests in functional repo
* ciGroup 10: 2 query enhancement tests in dashboard repo
* ciGroup 11: dashboard sanity test in dashboard repo

https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/12302916791

File: cypress/integration/apps/query_enhancements/dataset_selector.spec.js
Comment (d-rowe) [inline cypress/integration/apps/query_enhancements/dataset_selector.spec.js:20]: I know you're migrating these, but thinking that these type of hardcoded visits are good opportunities for utils in the future.

Ex.
```
DiscoverPage.visit({
    timeRange: {
        from: '2015-09-19T13:31:44.000Z',
        to: '2015-09-24T01:31:44.000Z'
    }
});
```

File: cypress/integration/apps/query_enhancements/dataset_selector.spec.js
Comment (d-rowe) [inline cypress/integration/apps/query_enhancements/dataset_selector.spec.js:22]: Related to comment above: this can then be rolled into the Discover page visit helper so that no other consumers need to worry about load timing.

Again, no need to address for this PR. Just sharing some thoughts for longterm changes.

**Review by d-rowe — COMMENTED** [review]
Nice! I know this migration work is surprisingly challenging.

Do we have a PR prepared to remove these from ftr or will that be a followup?

File: cypress/integration/apps/query_enhancements/dataset_selector.spec.js
Comment (ananzh) [reply cypress/integration/apps/query_enhancements/dataset_selector.spec.js:20]: Seems we are still using the data from data-explorer. Will we do a follow-up to update the data? This is the one for query_enhancement https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9006

Comment (abbyhu2000) [reply cypress/integration/apps/query_enhancements/dataset_selector.spec.js:20]: Sure, will do a follow up to update the data after 9006 is merged

**Comment by joshuali925** [comment]
do you have more background on why tests are moving from functional test repo to here? for example for other developers which repo should they use to add cypress tests

and is it necessary to add over one million lines of test data which doesn't appear to be from coming https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1661?

File: yarn.lock
Comment (sejli) [inline yarn.lock:1]: Is there a reason why a lot of dependencies were bumped? Are they necessary for the move from the `opensearch-dashboards-functional-test` repository to this one?

**Comment by abbyhu2000** [comment]
> do you have more background on why tests are moving from functional test repo to here? for example for other developers which repo should they use to add cypress tests
> 
> and is it necessary to add over one million lines of test data which doesn't appear to be from coming [opensearch-project/opensearch-dashboards-functional-test#1661](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1661)?

@joshuali925 I think the future plan is to have dashboard cypress tests within the dashboard repo, and write future dashboard cypress test in dashboard repo. So in the future when writing features, we can add corresponding cypress tests in the same PR, then it can directly test the feature running those tests in a single PR. (Do not need to raise separate PR in functional repo to verify).

This PR just starts the process of migrating some old cypress tests back to dashboard repo. 

cc: @ashwin-pc

**Comment by abbyhu2000** [comment]
> do you have more background on why tests are moving from functional test repo to here? for example for other developers which repo should they use to add cypress tests
> 
> and is it necessary to add over one million lines of test data which doesn't appear to be from coming [opensearch-project/opensearch-dashboards-functional-test#1661](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1661)?

both logstash data and discover data are data explorer plugin coming from functional repo. I will raise a follow up PR to clean the data and use the query enhancement data once this PR is merged: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9006

File: yarn.lock
Comment (abbyhu2000) [reply yarn.lock:1]: sorry yea yarn.lock changes should not be included here, will remove these changes

Comment (abbyhu2000) [reply yarn.lock:1]: removed

**Comment by abbyhu2000** [comment]
For some reasons, auto generate changelog mechanism does not work properly. Have manually added a changelog

**Review by ananzh — APPROVED** [review]

**Comment by ananzh** [comment]
@abbyhu2000 will do a follow up to update data and clean out. Approve this to unblock CI and tests merging.

**Review by sejli — APPROVED** [review]

**Comment by Hailong-am** [comment]
@ananzh can you do manual backport for this to 2.x?

**Pull request merged**
